physics/GFS_rrtmg_pre.F90

```diff
@@ -1053,6 +1057,38 @@ subroutine GFS_rrtmg_pre_run (im, levs, lm, lmk, lmp, n_var_lndp,        &
          enddo
        enddo
 
+! --- add spp
+      do k=1,levs
+        do i=1,im
+            pattern_spp_rad(i,k)=0.0
+        enddo
+      enddo
+
+      if ( do_spp ) then
+        do k=1,levs
+            do i=1,im
+                pattern_spp_rad(i,k)=spp_wts_rad(i,k)
+            enddo
+        enddo
+      endif
+
+      do k=1,lm
+        if (k < levs) then
+          do i=1,im
+            effrl_inout(i,k) = effrl_inout(i,k) - pattern_spp_rad(i,k)*5
+            effri_inout(i,k) = effri_inout(i,k) - pattern_spp_rad(i,k)*40
+            effrs_inout(i,k) = effrs_inout(i,k) - pattern_spp_rad(i,k)*80
+          enddo
+        else
+          do i=1,im
+            effrl_inout(i,k) = effrl_inout(i,k) - pattern_spp_rad(i,levs)*5
+            effri_inout(i,k) = effri_inout(i,k) - pattern_spp_rad(i,levs)*40
+            effrs_inout(i,k) = effrs_inout(i,k) - pattern_spp_rad(i,levs)*80
+          enddo
+        endif
+      enddo
+
+
```

Comment:
only one blank line between the new code and the existing code

Reply:
ya, will fix!